### DISCUSSION TITLE
clang-tidy against `test/`

### DISCUSSION THREAD

**Discussion by e-kwsm** [neovim/neovim#31653] [Show and tell]
I set `CMAKE_C_CLANG_TIDY` by myself and build nvim, and found that `test/` is not checked.
Is this intentional?

---

The following lines are not compliant with `llvm-include-order`:
https://github.com/neovim/neovim/blob/4e130c1ee446f4389a8c76c5e81b53bff8b9193c/test/unit/fixtures/multiqueue.c#L1-L4
which must be
```diff
-#include <string.h>
-#include <stdlib.h>
 #include "nvim/event/multiqueue.h"
 #include "multiqueue.h"
+#include <stdlib.h>
+#include <string.h>
```

---

There are mismatches of parameter names (`readability-inconsistent-declaration-parameter-name`):

https://github.com/neovim/neovim/blob/4e130c1ee446f4389a8c76c5e81b53bff8b9193c/test/unit/fixtures/multiqueue.h#L3 vs https://github.com/neovim/neovim/blob/4e130c1ee446f4389a8c76c5e81b53bff8b9193c/test/unit/fixtures/multiqueue.c#L7

and

https://github.com/neovim/neovim/blob/4e130c1ee446f4389a8c76c5e81b53bff8b9193c/test/unit/fixtures/multiqueue.h#L4 vs https://github.com/neovim/neovim/blob/4e130c1ee446f4389a8c76c5e81b53bff8b9193c/test/unit/fixtures/multiqueue.c#L12

**Comment by dundargoc** [comment]
Yeah, we basically never touch those files so didn't feel it necessary to include them. We can format test files if there is a need for it but it's probably not worth it since it'd involve extra cmake code.

**Reply by e-kwsm** [reply]
I see, thank you for your reply.

**Reply by dundargoc** [reply]
We can format those files once and be done with it.

**Reply by dundargoc** [reply]
https://github.com/neovim/neovim/pull/31668
